Tracy Allocator - build error

### Zig Version

0.11.0-dev.632+d69e97ae1

### Zig Language Server Version

ddf3ef1389e9f7c056e77ef808d6a746b8adbfc9

### Steps to Reproduce

Run build
```bash
zig build -Drelease-safe -Denable_tracy -Denable_tracy_allocation
```

### Expected Behavior

builded `zig-out/bin/zls`

### Actual Behavior

```bash
src/tracy.zig:149:37: error: root struct of file 'mem.Allocator' has no member named 'init'
            return std.mem.Allocator.init(self, allocFn, resizeFn, freeFn);
                   ~~~~~~~~~~~~~~~~~^~~~~
referenced by:
    main: src/main.zig:253:82
    callMain: /home/kassane/.local/lib/zig/std/start.zig:614:32
    remaining reference traces hidden; use '-freference-trace' to see all reference traces

error: zls...
error: The following command exited with error code 1:
/home/kassane/.local/bin/zig build-exe /home/kassane/Documentos/Projetos/zls/src/main.zig -cflags -DTRACY_ENABLE=1 -fno-sanitize=undefined -- /home/kassane/Documentos/Projetos/zls/src/tracy/TracyClient.cpp -lc++ -lc -OReleaseSafe --cache-dir /home/kassane/Documentos/Projetos/zls/zig-cache --global-cache-dir /home/kassane/.cache/zig --name zls --pkg-begin build_options /home/kassane/Documentos/Projetos/zls/zig-cache/options/w2qR1q79hioHq7w2xtUXpJMaJIB6gT9qYDogQKzmn9Qqr3A35Oy7x2NlQ_7vkWXz --pkg-end --pkg-begin known-folders /home/kassane/Documentos/Projetos/zls/src/known-folders/known-folders.zig --pkg-end -I /home/kassane/Documentos/Projetos/zls/src/tracy --enable-cache 
error: the following build command failed with exit code 1:
/home/kassane/Documentos/Projetos/zls/zig-cache/o/797fc7177a01e84bb7f8b12ca1f13bc7/build /home/kassane/.local/bin/zig /home/kassane/Documentos/Projetos/zls /home/kassane/Documentos/Projetos/zls/zig-cache /home/kassane/.cache/zig -Drelease-safe -Denable_tracy -Denable_tracy_allocation
```
**Reference:**
https://github.com/zigtools/zls/blob/ddf3ef1389e9f7c056e77ef808d6a746b8adbfc9/src/tracy.zig#L149

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -22,7 +22,7 @@ jobs:
         with:
           fetch-depth: 0
           submodules: true
-      - uses: goto-bus-stop/setup-zig@v1
+      - uses: goto-bus-stop/setup-zig@v2
         with:
           version: master
 
@@ -31,6 +31,9 @@ jobs:
 
       - name: Build
         run: zig build
+
+      - name: Build with Tracy
+        run: zig build -Denable_tracy -Denable_tracy_allocation
 
       - name: Run Tests
         run: zig build test

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -33,6 +33,7 @@ jobs:
         run: zig build
 
       - name: Build with Tracy
+        if: ${{ matrix.os != 'macos-latest' }}
         run: zig build -Denable_tracy -Denable_tracy_allocation
 
       - name: Run Tests

--- a/src/tracy.zig
+++ b/src/tracy.zig
@@ -146,44 +146,54 @@ pub fn TracyAllocator(comptime name: ?[:0]const u8) type {
         }
 
         pub fn allocator(self: *Self) std.mem.Allocator {
-            return std.mem.Allocator.init(self, allocFn, resizeFn, freeFn);
+            return .{
+                .ptr = self,
+                .vtable = &.{
+                    .alloc = allocFn,
+                    .resize = resizeFn,
+                    .free = freeFn,
+                },
+            };
         }
 
-        fn allocFn(self: *Self, len: usize, ptr_align: u29, len_align: u29, ret_addr: usize) std.mem.Allocator.Error![]u8 {
-            const result = self.parent_allocator.rawAlloc(len, ptr_align, len_align, ret_addr);
+        fn allocFn(ptr: *anyopaque, len: usize, ptr_align: u8, ret_addr: usize) ?[*]u8 {
+            const self = @ptrCast(*Self, @alignCast(@alignOf(Self), ptr));
+            const result = self.parent_allocator.rawAlloc(len, ptr_align, ret_addr);
             if (result) |data| {
-                if (data.len != 0) {
+                if (len != 0) {
                     if (name) |n| {
-                        allocNamed(data.ptr, data.len, n);
+                        allocNamed(data, len, n);
                     } else {
-                        alloc(data.ptr, data.len);
+                        alloc(data, len);
                     }
                 }
-            } else |_| {
+            } else {
                 messageColor("allocation failed", 0xFF0000);
             }
             return result;
         }
 
-        fn resizeFn(self: *Self, buf: []u8, buf_align: u29, new_len: usize, len_align: u29, ret_addr: usize) ?usize {
-            if (self.parent_allocator.rawResize(buf, buf_align, new_len, len_align, ret_addr)) |resized_len| {
+        fn resizeFn(ptr: *anyopaque, buf: []u8, buf_align: u8, new_len: usize, ret_addr: usize) bool {
+            const self = @ptrCast(*Self, @alignCast(@alignOf(Self), ptr));
+            if (self.parent_allocator.rawResize(buf, buf_align, new_len, ret_addr)) {
                 if (name) |n| {
                     freeNamed(buf.ptr, n);
-                    allocNamed(buf.ptr, resized_len, n);
+                    allocNamed(buf.ptr, new_len, n);
                 } else {
                     free(buf.ptr);
-                    alloc(buf.ptr, resized_len);
+                    alloc(buf.ptr, new_len);
                 }
 
-                return resized_len;
+                return true;
             }
 
             // during normal operation the compiler hits this case thousands of times due to this
             // emitting messages for it is both slow and causes clutter
-            return null;
+            return false;
         }
 
-        fn freeFn(self: *Self, buf: []u8, buf_align: u29, ret_addr: usize) void {
+        fn freeFn(ptr: *anyopaque, buf: []u8, buf_align: u8, ret_addr: usize) void {
+            const self = @ptrCast(*Self, @alignCast(@alignOf(Self), ptr));
             self.parent_allocator.rawFree(buf, buf_align, ret_addr);
             // this condition is to handle free being called on an empty slice that was never even allocated
             // example case: `std.process.getSelfExeSharedLibPaths` can return `&[_][:0]u8{}`

